pkg/cmd/admin/policy/cani.go

```diff
@@ -209,13 +226,18 @@ func (o *canIOptions) listAllPermissions() error {
 		if err != nil {
 			return err
 		}
-		rulesReviewStatus = whatCanYouDo.Status
 
+		policyRules = whatCanYouDo.Status.Rules
+		rulesReviewResult = whatCanYouDo
+	}
+
+	if o.Printer != nil {
+		return o.Printer.PrintObj(rulesReviewResult, o.Out)
 	}
 
 	writer := tabwriter.NewWriter(o.Out, tabwriterMinWidth, tabwriterWidth, tabwriterPadding, tabwriterPadChar, tabwriterFlags)
```

Comment:
@fabianofranz not sure TBH. Maybe @juanvallejo can tell us more. One thing that I saw at the docs is the fact that `HumanReadablePrinter` is not threadsafe. Should we be concerned about it?

What would be the practical benefit of using it? Get rid of `AddPrinterFlags` ?